week-4-airflow/airflow-home/dags/train_pipeline.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+
+from airflow import DAG
+from airflow.models import Variable
+from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
+from kubernetes.client import models as k8s
+
+
+volume = k8s.V1Volume(name="training-storage", persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name="training-storage"),)
+volume_mount = k8s.V1VolumeMount(name="training-storage", mount_path="/tmp", sub_path=None)
+
+
+# -------------
+# TODO: Investigate https://airflow.apache.org/docs/apache-airflow-providers-cncf-kubernetes/2.0.3/_modules/airflow/providers/cncf/kubernetes/example_dags/example_kubernetes.html
+# -------------
+
+DATASET_PATH = "/tmp/dataset.json"
+SCORES_PATH = "/tmp/scores.json"
+MODEL_PATH = "/tmp/cls.pekl"
+
+WANDB_PROJECT = Variable.get("WANDB_PROJECT")
+WANDB_API_KEY = Variable.get("WANDB_API_KEY")
+
+with DAG(
+    description="Airflow Training on K8S", 
+    catchup=False, 
+    dag_id="training_dag",
+    start_date=datetime(2021, 1, 1),
+    schedule=None,
+    tags=["K8S", "Course"]) as dag:
+
+    clean_storage_before_start = KubernetesPodOperator(
+        name="clean_storage_before_start",
+        image="bash:latest",
+        cmds=["rm", "-rf", "/tmp/*"],  # actually it doesn't work as expected
```

Comment:
why doesn't as expected work?

Reply:
Probably (my hypothesis) because of how minikube internally manages persistent volumes.

The command:
```
cmds=["rm", "-rf", "/tmp/"]
```
makes pipeline fail with the message: 
```
rm: cannot remove '/tmp/': Device or resource busy
```


The command:
```
cmds=["rm", "-rf", "/tmp/*"]
```
actually don't remove anything.


The only proper way to remove is to remove each particular file:
```
cmds=["rm", "-rf", DATASET_PATH, SCORES_PATH, MODEL_PATH],
```